src/engraving/accessibility/accessibleroot.cpp

```diff
@@ -170,3 +171,64 @@ void AccessibleRoot::setMapToScreenFunc(const AccessibleMapToScreenFunc& func)
 {
     m_accessibleMapToScreenFunc = func;
 }
+
+bool AccessibleRoot::isRangeSelection() const
+{
+    if (auto focusedItem = m_focusedElement.lock()) {
+        const EngravingItem* item = focusedItem->element();
+        if (item) {
+            return item->score()->selection().isRange();
+        }
+    }
+    return false;
+}
+
+QString AccessibleRoot::rangeSelectionInfo()
+{
+    auto focusedItem = m_focusedElement.lock();
+    if (!focusedItem) {
+        return QString();
+    }
+    const EngravingItem* item = focusedItem->element();
+    if (!item) {
+        return QString();
+    }
+
+    Selection selection = item->score()->selection();
+    Segment* startSegment = selection.startSegment();
+    Segment* endSegment = selection.endSegment();
+    if (!startSegment || !endSegment) {
+        return QString();
+    }
+    String start = startSegment->formatBarsAndBeats();
+    String end;
+    if (Segment* endSeg = endSegment->prev1(SegmentType::ChordRest)) {
+        end = endSeg->formatBarsAndBeats();
+        end.remove(u';'); // Too many pauses in speech
+    }
+
+    start.remove(u';');
+
+    QString staffInstrument1, staffInstrument2;
+    staff_idx_t startStaff = selection.staffStart();
+    staff_idx_t endStaff = selection.staffEnd() - 1;
+
+    if (startStaff != endStaff) {
+        Staff* staff1 = m_element->score()->staff(startStaff);
+        Staff* staff2 = m_element->score()->staff(endStaff);
```

Comment:
Remember to use the new `score` variable here instead of calling `m_element->score()`